### PR TITLE
ANALYTICS-4522 | RARS: Migrate top_referrers endpoint to RARS

### PR DESCRIPTION
**Pull request by Jordan3900**
**References**: [ANALYTICS-4522](https://jira.gannett.com/browse/ANALYTICS-4522)

**Code PR**: https://github.com/GannettDigital/reach-analytics-reporting-service/pull/1706

**Description**:
A few sentences describing the overall goals of the pull request's commits.

### CODE REVIEW

File: source/includes/_rars_top_referrers.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                        "count": 9
+                    }
+                ]
+            }
+        ]
+    }
+}
+```
+
+The body of the API response is an array of source objects.
```
Comment (NBnfork) [inline source/includes/_rars_top_referrers.md:128]: This section is a little unclear. Can you update the headers and rearrange it like this?
```
The main body of the API response is found in the report_data object.
**Base Object**
|Field Name|Datatype|Description|
|---|---|---|
|api_name|String|Name of the API|
|api_run_date|String|Date report was run|
|start_date|String|Start date of report|
|end_date|String|End date of report|
|time_zone|String|Time Zone|
|interval_size|String|Interval Size ('none', 'day', 'calendar_week' or 'calendar_month')|
|report_data|Object|[Report Data Object](#topreferrerreportdata)

<a id="topreferrerreportdata"></a>
**Report Data Object**
Field Name | Datatype | Nullable | Description
---------- | -------- | -------- | -----------
sources | Array | Yes | An array of source objects

**Source Object**
Field Name | Datatype | Nullable | Description
---------- | -------- | -------- | -----------
name | String | Yes | The type of campaign SEARCH, OTHER, DIRECT
master_campaign_id | String | Yes | The master campaign id
is_paid | boolean | No | Is the campaign a paid campaign
top_referrers | Array | No | An array of [top referrer objects](#array-of-top-referrers)

<a id="array-of-top-referrers"></a>
**Array of top referrers**
Field Name | Datatype | Nullable | Description
---------- | -------- | -------- | -----------
referrer_host | String | No | The hostname
count | Integer | No | The count of referrals from the specified host
```

**Review by NBnfork — CHANGES_REQUESTED** [review]
I think the bottom section was a little unclear.

**Review by NBnfork — APPROVED** [review]

File: source/includes/_rars_top_referrers.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+### Parameters&nbsp;
+
+When using the GET method, the results can be filtered using these parameters:
+
+| Param     | Required | Function |
+|-----------|-----|---|
+| start_date | Yes | Restricts the results to those occurring on or after this date and time. |
+| end_date | Yes | Restricts the results to those occurring on or before this date and time. |
+| gmaid | Yes | Restrict results to the specified advertiser |
```
Comment (ezetter) [inline source/includes/_rars_top_referrers.md:21]: This is inconsistent with the gmaid being part of the URL, as described above. I'm assuming the intent is to make this part of the URL, consistent with Marketing Activity.

File: source/includes/_rars_top_referrers.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    "start_date": "2022-11-10",
+    "end_date": "2022-11-15",
+    "time_zone": "America/Los_Angeles",
+    "interval_size": "calendar_week",
+    "global_master_advertiser_id": "TEST_1",
+    "report_data": {
+        "sources": [
+            {
+                "name": "DIRECT",
+                "master_campaign_id": 1,
```
Comment (ezetter) [inline source/includes/_rars_top_referrers.md:48]: By definition DIRECT traffic won't have a MCID associated. Will this be null in that case (apparently that's how it was with the old API), or missing?

File: source/includes/_rars_top_referrers.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+### Parameters&nbsp;
+
+When using the GET method, the results can be filtered using these parameters:
+
+| Param     | Required | Function |
+|-----------|-----|---|
+| start_date | Yes | Restricts the results to those occurring on or after this date and time. |
+| end_date | Yes | Restricts the results to those occurring on or before this date and time. |
+| gmaid | Yes | Restrict results to the specified advertiser |
+|interval_size|No|Return subtotals for the given intervals within the start-end time provided. Accepts calendar_month, calendar_week, or day (Default is day)|
```
Comment (NBnfork) [inline source/includes/_rars_top_referrers.md:22]: @Jordan3900 we are not using interval_size in the report building process. So it should not be a param here.

**Pull request merged**